src/jit/morph.cpp

```diff
@@ -2291,17 +2291,18 @@ void fgArgInfo::EvalArgsToTemps()
                     if (setupArg->OperIsCopyBlkOp())
                     {
                         setupArg = compiler->fgMorphCopyBlock(setupArg);
-#if defined(_TARGET_ARMARCH_)
-                        // This scalar LclVar widening step is only performed for ARM architectures.
+#if defined(_TARGET_ARMARCH_) || defined(UNIX_AMD64_ABI)
+                        // This scalar LclVar widening step is only performed for ARM and AMD64 unix.
```

Comment:
Seems like if we fixed `getArgTypeForStruct` to return `SPK_EnclosingType` for the odd size cases, and then called that here instead of `getPrimitiveTypeForStruct`, then getting back `SPK_EnclosingType` would tell us when we need to widen.

Reply:
I agree that at a minimum this should be better commented. That being said, I think this should be addressed later and I will merge now to unblock ppl asap.